src/Microsoft.Health.Dicom.SqlServer/Features/CustomTag/SqlCustomTagStore.cs

```diff
@@ -87,9 +88,36 @@ public Task<IEnumerable<CustomTagEntry>> GetAllCustomTagsAsync(CancellationToken
             throw new System.NotImplementedException();
         }
 
-        public Task DeleteCustomTagAsync(string tagPath, string vr, CancellationToken cancellationToken = default)
+        public async Task DeleteCustomTagAsync(string tagPath, string vr, CancellationToken cancellationToken = default)
         {
-            throw new System.NotImplementedException();
+            if (_schemaInformation.Current < SchemaVersionConstants.SupportCustomTagSchemaVersion)
+            {
+                throw new BadRequestException(DicomSqlServerResource.SchemaVersionNeedsToBeUpgraded);
+            }
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapperAsync(cancellationToken))
+            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                VLatest.DeleteCustomTag.PopulateCommand(sqlCommandWrapper, tagPath, (byte)CustomTagLimit.CustomTagVRAndDataTypeMapping[vr]);
+
+                try
+                {
+                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
+                }
+                catch (SqlException ex)
+                {
+                    switch (ex.Number)
+                    {
+                        case SqlErrorCodes.NotFound:
+                            throw new CustomTagNotFoundException();
+                        case SqlErrorCodes.PreconditionFailed:
+                            throw new PreconditionFailedException(
+                                string.Format(CultureInfo.InvariantCulture, DicomSqlServerResource.InvalidCustomTagStatus, tagPath));
```

Comment:
Given that the deletions are not jobs yet, the status is always `1`, right? Although it's good to handle this case before the implementation is in place!

However, I just figured out that we're sort of piggy-backing off of HTTP status codes, right? It's like the mandatory 50,000 range + error code. So, I would probably not make an association to the precondition failed error code, only because that is pretty specific to the HTTP headers, as per [Mozilla](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/412). This error is really indicating that you can't delete this custom tag because it is being processed (presumably it is mid-index or mid-delete right?). 

So I would probably say either come up with a tag-specific error like `CustomTagBusyException` or something more generic that's built-in like `InvalidOperationException` to convey this

Reply:
will change to CustomTagBusyException